gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/ExploreService.java

```diff
@@ -102,6 +109,21 @@ public ExploreResponse explore(RequestContext requestContext, ExploreRequest req
           requestHandler.handleRequest(newExploreRequestContext, request);
 
       return responseBuilder.build();
+    } catch (Exception ex) {
+      boolean acquireSucceeded = rateLimiter.tryAcquire();
```

Comment:
Check debug before trying to acquire rate limiter (which is an expensive operation).